src/components/TextInputWithCurrencySymbol/index.android.tsx

```diff
@@ -0,0 +1,48 @@
+import React, {useEffect, useState} from 'react';
+import type {NativeSyntheticEvent, TextInput, TextInputSelectionChangeEventData} from 'react-native';
+import BaseTextInputWithCurrencySymbol from './BaseTextInputWithCurrencySymbol';
+import type TextInputWithCurrencySymbolProps from './types';
+
+function TextInputWithCurrencySymbol(
+    {
+        selectedCurrencyCode,
+        onCurrencyButtonPress = () => {},
+        onChangeAmount = () => {},
+        formattedAmount,
+        placeholder,
+        selection,
+        onSelectionChange = () => {},
+        onKeyPress = () => {},
+    }: TextInputWithCurrencySymbolProps,
+    ref: React.ForwardedRef<TextInput>,
+) {
+    const [skipNextSelectionChange, setSkipNextSelectionChange] = useState(false);
+
+    useEffect(() => {
+        setSkipNextSelectionChange(true);
+    }, [formattedAmount]);
+
+    return (
+        <BaseTextInputWithCurrencySymbol
+            ref={ref}
+            onCurrencyButtonPress={onCurrencyButtonPress}
+            onChangeAmount={onChangeAmount}
+            formattedAmount={formattedAmount}
+            placeholder={placeholder}
+            selectedCurrencyCode={selectedCurrencyCode}
+            onKeyPress={onKeyPress}
+            selection={selection}
```

Comment:
Maybe we can leave spread props the same way as it was before, just in case, wdyt?

So
```
function TextInputWithCurrencySymbol(
    {onSelectionChange = () => {}, ...props}: TextInputWithCurrencySymbolProps,
    ref: React.ForwardedRef<TextInput>,
)
```
and then
```
 <BaseTextInputWithCurrencySymbol
            /* eslint-disable-next-line react/jsx-props-no-spreading */
            {...props}
            ref={ref}
            onSelectionChange={(event: NativeSyntheticEvent<TextInputSelectionChangeEventData>) => {
                if (skipNextSelectionChange) {
                    setSkipNextSelectionChange(false);
                    return;
                }
                onSelectionChange(event);
            }}
        />
```

Reply:
I agree with you 

I think this is better since we need here to pass only two props additionally